app/models/ext_management_system.rb

```diff
@@ -84,11 +84,16 @@ def self.supported_types_and_descriptions_hash
   def hostname_uniqueness_valid?
     return unless hostname_required?
     return unless hostname.present? # Presence is checked elsewhere
-    # check uniqueness per provider type
-
-    existing_hostnames = (self.class.all - [self]).map(&:hostname).compact.map(&:downcase)
-
-    errors.add(:hostname, N_("has to be unique per provider type")) if existing_hostnames.include?(hostname.downcase)
+    hostname_without_slash = hostname.downcase.gsub(/\/$/, '')
+    # get all hostnames on database without '/' at the end
+    existing_hostnames = (self.class.all - [self]).map(&:hostname).compact.map { |hostname| hostname.downcase.gsub(/\/$/, '') }
+    errors.add(:hostname, N_("has to be unique per provider type")) if existing_hostnames.include?(hostname_without_slash)
+
+    # if hostname is ipaddress check against dicovered ipaddress in database too
+    if hostname.ipaddress?
+      existing_ipaddress = (self.class.all - [self]).map(&:ipaddress).compact.map { |ipaddress| ipaddress.gsub(/\/$/, '') }
+      errors.add(:ipaddress, N_("has to be unique per provider type")) if existing_ipaddress.include?(hostname_without_slash)
+    end
```

Comment:
Under what circumstance would there be an EMS with an `:ipaddress` property that isn't equal to the `:hostname`?

Reply:
You can bind a provider by the hostname or the ip address, and the ip address field always will be filled with the IP of the provider, in summary if the `hostname` is an ip address both `hostname` and `ipaddress` will be equals, else, will be differents. So when try to add a new provider by its ip address, the system must verify against `hostname` and `ipaddress` on the DB to ensure that will avoid duplicated providers.

Reply:
> and the ip address field always will be filled with the IP of the provider

Is this something the lenovo provider does?  Because I don't think this is true across the board, I just tested with a VMware provider through the UI and the ipaddress wasn't filled in.

I also tested creating a provider with just an ipaddress (and no hostname) through a rails console and it threw an exception:
`manageiq/app/models/ext_management_system.rb:99:in `hostname_format_valid?': undefined method `ipaddress?' for nil:NilClass (NoMethodError)`

Reply:
According with my tests, seems like that others providers doesn't fill the `ipaddress` field on the refresh routine, however all providers has this field, once that it is attached to an `ExtManagementSystem` and is shown on view through the "Discovered IP address" field on provider's table, as we can see on the image below:
![image](https://user-images.githubusercontent.com/8550928/36270995-0978b1ec-125c-11e8-954f-2f6558d73c69.png)
In the particular case of Lenovo provider, the IP address is filled even the user regist the provider through its hostname, as we can see below:
![image](https://user-images.githubusercontent.com/8550928/36271105-65e30e3c-125c-11e8-9b13-6821ae094767.png)
So, without this verification added on this PR the user would be able to add the provider 10.243.9.123 (for example) again, if inform only its IP address on the hostname input.